### ISSUE TITLE
Memory leak caused both version 2.0.1 and 2.0.2 when using cache

### ISSUE THREAD

**Issue by nguyenhuybo** [opensagres/xdocreport#366]
Calling loadReport about 500 times
---------------->VERSION 2.0.1 <------------------

IXDocReport report = XDocReportRegistry.getRegistry().loadReport(in, TemplateEngineKind.Velocity);  caused Memory leak

Because loadReport() default cached template
return loadReport( sourceStream, templateEngineKind, true );  <--- cache

--------------> VERSION 2.0.2 <--------------------
IXDocReport report = XDocReportRegistry.getRegistry().loadReport(in, TemplateEngineKind.Velocity); 
 default cache is false
return loadReport( sourceStream, templateEngineKind, false ); <--- not cache anymore and not Memory leak

But to use cache then do the following
IXDocReport report = XDocReportRegistry.getRegistry().loadReport(in, TemplateEngineKind.Velocity, true);

This time the template is still cached but still caused MEMORY LEAK

Thank you :-D
![image](https://user-images.githubusercontent.com/10195703/57688896-b7923700-7668-11e9-9677-51f9b5b783f8.png)

_Originally posted by @nguyenhuybo in https://github.com/opensagres/xdocreport/issues/229#issuecomment-492166413_

**Comment by angelozerr** [comment]
Yes it's normal, when you call loadReport it loads report and cache it. The InpuStream cannot be used as key for cache. You must use loadReport the first time and getReport the second time.

Please read https://github.com/opensagres/xdocreport/wiki/DocxReportingJavaMain#test-performance

**Comment by nguyenhuybo** [comment]
Hi Angelo

My financial service system uses so many different reports and generates one report many times then I decided to use caching. 
But maybe someday the cached data is huge. Then I decided to use  setClearTimeout( long timeout )

But I do not understand your explanation
/**
	 * Clear cache with timeout. If this method is called, you must call
	 * {@link XDocReportRegistry#dispose()} method at the end of your program to
	 * stop the timer Thread.
	 * 
	 * @param timeout
	 */
    public void setClearTimeout( long timeout )
